src/OrleansProviders/Streams/Common/PooledCache/ObjectPool.cs

```diff
@@ -52,7 +82,15 @@ public virtual T Allocate()
         /// <param name="resource"></param>
         public virtual void Free(T resource)
         {
+            this.monitor?.TrackObjectReleasedFromCache(1);
             pool.Push(resource);
         }
+
+        private void ReportObjectPoolStatistics(object state)
+        {
+            var freeBlocks = this.pool.Count;
+            long claimedBlocks = this.totalBlocks - freeBlocks;
+            this.monitor.Report(this.totalBlocks, freeBlocks, claimedBlocks);
```

Comment:
I think availableBlocks may be more clear than free blocks.  Free usually means returned to the OS.